components/script/dom/canvasrenderingcontext2d.rs

```diff
@@ -36,12 +36,35 @@ use crate::dom::path2d::Path2D;
 use crate::dom::textmetrics::TextMetrics;
 use crate::script_runtime::CanGc;
 
+#[derive(JSTraceable, MallocSizeOf)]
+struct DroppableCanvasRenderingContext2D {
+    pub canvas_state: CanvasState,
+}
+
+impl DroppableCanvasRenderingContext2D {
+    pub(crate) fn new(canvas_state: CanvasState) -> Self {
+        DroppableCanvasRenderingContext2D { canvas_state }
+    }
+}
+
+impl Drop for DroppableCanvasRenderingContext2D {
+    fn drop(&mut self) {
+        if let Err(err) = self
+            .canvas_state
+            .get_ipc_renderer()
```

Comment:
Since the only part of CanvasState that is used in the Drop implementation is the renderer channel, we can simplify this code by only storing a clone of the channel in the DroppableCanvasRenderingContext2D, rather than the entire CanvasState. This also avoids the error you see about CanvsState requiring rooting.

Reply:
I actually mean cloning the reference so you are not trying to store a reference.

Reply:
@jdm I did it!